clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -547,16 +580,29 @@ public void onFailure(RuntimeException e) {
      *
      * @param node The node to send the ListOffsetRequest to.
      * @param timestampsToSearch The mapping from partitions to the target timestamps.
+     * @param requireTimestamp  True if we require a timestamp in the response.
      * @return A response which can be polled to obtain the corresponding timestamps and offsets.
      */
     private RequestFuture<Map<TopicPartition, OffsetAndTimestamp>> sendListOffsetRequest(final Node node,
-                                                                                         final Map<TopicPartition, Long> timestampsToSearch) {
-        ListOffsetRequest request = new ListOffsetRequest(timestampsToSearch, ListOffsetRequest.CONSUMER_REPLICA_ID);
-        log.trace("Sending ListOffsetRequest {} to broker {}", request, node);
-        return client.send(node, ApiKeys.LIST_OFFSETS, request)
+                                                                                         final Map<TopicPartition, Long> timestampsToSearch,
+                                                                                         boolean requireTimestamp) {
+        ListOffsetRequest.Builder builder = new ListOffsetRequest.Builder().
+                setTargetTimes(timestampsToSearch).
+                setReplicaId(ListOffsetRequest.CONSUMER_REPLICA_ID);
+
+        // If we need a timestamp in the response, the minimum RPC version we can send is v1.
+        // Otherwise, v0 is OK.
+        builder.setMinVersion(requireTimestamp ? (short) 1 : (short) 0);
+
+        log.trace("Sending ListOffsetRequest {} to broker {}", builder, node);
+        return client.send(node, builder)
                 .compose(new RequestFutureAdapter<ClientResponse, Map<TopicPartition, OffsetAndTimestamp>>() {
                     @Override
                     public void onSuccess(ClientResponse response, RequestFuture<Map<TopicPartition, OffsetAndTimestamp>> future) {
+                        if (response.versionMismatch() != null) {
```

Comment:
Right, I have been looking at that.  Thinking of doing it in a follow-on JIRA since I will need to take a deeper look at the RPC paths when doing that